tsl/test/expected/correlated_constraints.out

```diff
@@ -0,0 +1,285 @@
+SELECT compress_chunk(:'CH_NAME');
+WARNING:  no index on "sensor_id" found for correlated constraint on chunk "_hyper_1_1_chunk"
+             compress_chunk             
+----------------------------------------
+ _timescaledb_internal._hyper_1_1_chunk
+(1 row)
+
+-- check chunk compression status
+SELECT chunk_status
+FROM compressed_chunk_info_view
+WHERE hypertable_name = 'sample_table' AND chunk_name = :'CH_NAME';
+ chunk_status 
+--------------
+            1
+(1 row)
+
+-- There should be an entry with min/max range computed for this chunk
+SELECT * FROM _timescaledb_catalog.dimension_slice WHERE dimension_id = :dimension_id;
+ id | dimension_id |     range_start      |      range_end      
+----+--------------+----------------------+---------------------
+  3 |            2 | -9223372036854775808 | 9223372036854775807
+  4 |            2 |                    1 |                   9
+(2 rows)
+
+SELECT * FROM _timescaledb_catalog.chunk_constraint WHERE constraint_name LIKE '_$CC_con%' ORDER BY chunk_id;
+ chunk_id | dimension_slice_id |  constraint_name  | hypertable_constraint_name 
+----------+--------------------+-------------------+----------------------------
+        1 |                  4 | _$CC_constraint_4 | 
+        2 |                  3 | _$CC_constraint_3 | 
+(2 rows)
+
+-- A query using a WHERE clause on "sensor_id" column will scan the proper chunk
+-- due to chunk exclusion using correlated constraints ranges calculated above
+:PREFIX SELECT * FROM sample_table WHERE sensor_id > 9;
+          QUERY PLAN          
+------------------------------
+ Seq Scan on _hyper_1_2_chunk
+   Filter: (sensor_id > 9)
+(2 rows)
+
+-- Newly added chunks should also point to this MIN/MAX entry
+\set start_date '2024-01-28 01:09:51.583252+05:30'
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 1, 9.6054, 78.999, 'new row4');
+SELECT * FROM _timescaledb_catalog.chunk_constraint WHERE dimension_slice_id = :slice_id ORDER BY chunk_id;
+ chunk_id | dimension_slice_id |  constraint_name  | hypertable_constraint_name 
+----------+--------------------+-------------------+----------------------------
+        2 |                  3 | _$CC_constraint_3 | 
+        5 |                  3 | _$CC_constraint_3 | 
+(2 rows)
+
+-- use the remove_dimension API to remove the correlated dimension entries
+SELECT * from remove_dimension('sample_table', 'sensor_id');
+ dimension_id | dropped 
+--------------+---------
+            2 | t
+(1 row)
+
+SELECT * FROM _timescaledb_catalog.dimension WHERE type = 'C';
+ id | hypertable_id | column_name | column_type | aligned | num_slices | partitioning_func_schema | partitioning_func | interval_length | compress_interval_length | integer_now_func_schema | integer_now_func | type 
+----+---------------+-------------+-------------+---------+------------+--------------------------+-------------------+-----------------+--------------------------+-------------------------+------------------+------
```

Comment:
Maybe select on the dimension_id returned by `remove_dimension`? Or join in same query?